SIPS/sip-keeper-system.md

```diff
@@ -0,0 +1,163 @@
+---
+sip: 72
+title: Synthetix Keeper System
+status: WIP
+author: Jackson Chan (@jacko125)
+discussions-to: https://discord.gg/kPPKsPb
+
+created: 2020-07-14
+---
+
+## Simple Summary
+A generalised Synthetix keeper contract will be able to incentivise certain functions in the Synthetix protocol to be managed in a decentralised manner. The rewards, which can be SNX or other tokens / ether will be held in the keeper contract and the rewards schedule managed by the keeper contract.
+
+## Abstract
+This SIP proposes a generalised Synthetix keeper contract to reward keeper functions like Exchange settlement, iSynth freezing, Chainlink pull request for prices and Binary option market resolutions.
+
+## Motivation
+The reason for a generalised keeper system / contract is to provide reward incentives for decentralised keepers to help manage the Synthetix protocol. The generalised keeper system will be able to set the amount of incentive and rewards for each function in the platform that requires keepers to manage.
+
+As the platform moves towards using decentralized oracle feeds from Chainlink, it will require keepers to manage the freezing and purging of iSynths (as per [SIP-61](./sip-61.md)).
+
+Another important function that requires keepers is to settle fee reclamation Exchange entries if they are not settled during the oridinary exchange process of a user's trading. Unsettled fee reclamations create an in-determinate state on the system's debt pool once they can be settled after the waiting period or next price is available.
+
+Fee clamation can also be broken in the case that Chainlink's aggregators upgrade their contracts and the `roundID` values are reset, so that the efficient settlement of exchanges ensure that the debt pool is up to date and won't cause
+
+The reward incentives should cover the gas costs of executing the transaction in an efficient timely manner, ie using a 'fast' gas price to execute the transaction and be profitable for the keeper to continue operating.
+
+## Specification
+
+# Overview
+There are a number of core functions on Synthetix protocol that require keepers to perform and execute in a decentralised manner. In order to incentivise these keeper functions and cover the gas costs for the on-chain transactions, a generalised keeper rewards contract will provide reward incentives to keepers.
+
+### Actions
```

Comment:
Detailing these potential seems out of scope here - they may well change in the future (meaning this SIP will become stale) and frankly aren't critical to outline in detail for this SIP. I'd just summarize the current use-cases: `Synthetix.mint`, `Exchanger.settle` and `ExchangeRates.freeze` (TBD). That said, if you want to detail different features then it's fine to add a sentence or two, but be careful to not distract from the general feature you're proposing by going into depth about the various implementation reasons.

Reply:
I still think these are out of scope and should simply be left as summaries